servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.evictor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.Pair;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.NotFoundException;
+import tachyon.worker.block.BlockMetadataManagerView;
+import tachyon.worker.block.BlockStoreEventListenerBase;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageDirView;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.StorageTierView;
+
+/**
+ * This class is used to evict blocks by LRFU. LRFU combines LRU and LFU, it evicts blocks with
+ * small frequency or large recency. Actually, LRFU evicts blocks with minimum CRF. CRF of a block
+ * is the sum of F(t) = pow((1.0 / {@link #mAttenuationFactor}, t * {@link #mStepFactor}) and each
+ * access to a block has a F(t) value where t is the time interval since that access time to 
+ * current. When {@link #mStepFactor} is close to 0, LRFU is close to LFU. Conversely, LRFU is 
+ * close to LRU when {@link #mStepFactor} is close to 1
+ */
+public class LRFUEvictor extends BlockStoreEventListenerBase implements Evictor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static final double DEFAULT_STEP_FACTOR = 0.25;
+  private static final double DEFAULT_ATTENUATION_FACTOR = 2.0;
+
+  private BlockMetadataManagerView mManagerView;
+  // Map from block id to the last access time of the block
+  private Map<Long, Long> mBlockIdToLastAccessTime = new ConcurrentHashMap<Long, Long>();
```

Comment:
Please comment that the AccessTime is the timestamp in logic time counts (is it?)